internal/verifier/check.go

```diff
@@ -122,18 +122,20 @@ func (verifier *Verifier) CheckWorker(ctx context.Context) error {
 	return nil
 }
 
-func (verifier *Verifier) CheckDriver(ctx context.Context, testChan ...chan struct{}) error {
+func (verifier *Verifier) CheckDriver(ctx context.Context, filter bson.D, testChan ...chan struct{}) error {
 	verifier.mux.Lock()
 	if verifier.running {
 		verifier.mux.Unlock()
 		verifier.logger.Info().Msg("Verifier already checking the collections")
 		return nil
 	}
 	verifier.running = true
+	verifier.globalFilter = filter
 	verifier.mux.Unlock()
 	defer func() {
 		verifier.mux.Lock()
 		verifier.running = false
+		verifier.globalFilter = nil
```

Comment:
It's unset so that the filter is only used in the context of the current running of `CheckDriver` method. The next call of `CheckDriver` might run with a different `globalFilter`.